### PR TITLE
zk session timeout the same as commit timeout

### PR DESCRIPTION
**Pull request by adyach**
# One-line summary
Zookeeper session timeout the same as commit timeout

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/repository/zookeeper/ZooKeeperHolder.java [hunk 1]
```diff
@@ -65,12 +71,38 @@ private void initExhibitor() throws Exception {
                 .connectionTimeoutMs(connectionTimeoutMs)
                 .build();
         zooKeeper.start();
+
+        subscriptionCurator = CuratorFrameworkFactory.builder()
+                .ensembleProvider(ensembleProvider)
+                .retryPolicy(retryPolicy)
+                // max commit timeout is not higher than 60 seconds, it is safe to cast yo integer
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/repository/zookeeper/ZooKeeperHolder.java]: Yo!

File: src/main/java/org/zalando/nakadi/repository/zookeeper/ZooKeeperHolder.java [hunk 2]
```diff
@@ -65,12 +71,38 @@ private void initExhibitor() throws Exception {
...
+    public CuratorFramework getSubscriptionCurator(final long sessionTimeoutMs) {
+        // most of the clients use default max timeout, subscriptionCurator client saves zookeeper resource
+        if (sessionTimeoutMs == maxCommitTimeoutMs) {
+            return subscriptionCurator;
+        }
+
+        final CuratorFramework curatorFramework = CuratorFrameworkFactory.builder()
+                .connectString(zooKeeper.getZookeeperClient().getCurrentConnectionString())
+                .retryPolicy(new ExponentialBackoffRetry(EXHIBITOR_RETRY_TIME, EXHIBITOR_RETRY_MAX))
+                // max commit timeout is not higher than 60 seconds, it is safe to cast yo integer
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/repository/zookeeper/ZooKeeperHolder.java]: Yo!

**Comment by v-stepanov** [comment]
Everything seems to look good. I'm almost ready to approve.

**Comment by v-stepanov** [comment]
👍

**Comment by adyach** [comment]
👍

**Pull request merged**